lib/src/local_working_copy.rs

```diff
@@ -437,17 +440,42 @@ impl<'a> IntoIterator for FileStates<'a> {
     }
 }
 
-#[derive(Default, Clone, Debug)]
+#[derive(Clone, Debug)]
 #[non_exhaustive]
 /// Options that controls the behavior of the [`TreeState`] created.
-pub struct TreeStateSettings;
+pub struct TreeStateSettings {
+    /// Configuring auto-converting CRLF line endings into LF when you add a
+    /// file to the backend, and vice versa when it checks out code onto your
+    /// filesystem.
+    pub eol_conversion_settings: EolConversionSettings,
+}
 
 impl TreeStateSettings {
     /// Create [`TreeStateSettings`] from [`UserSettings`].
     pub fn try_from_user_settings(
-        _user_settings: &UserSettings,
+        user_settings: &UserSettings,
     ) -> Result<Self, impl Error + Send + Sync> {
-        Ok::<_, Infallible>(Self)
+        #[expect(
+            clippy::question_mark,
+            reason = "Can't use ? here, because the error type is an implicit type."
+        )]
+        let eol_conversion_settings =
+            match EolConversionSettings::try_get_from_settings(user_settings) {
+                Ok(value) => value,
+                Err(e) => return Err(e),
+            };
+        Ok(Self {
+            eol_conversion_settings,
+        })
+    }
+}
+
+impl Default for TreeStateSettings {
```

Comment:
nit: `#[default]` could be implemented on the eol enum, but I'm fine with either way.